![Brand](data:image/png;base64...)

* [Summary](#Summary)
* [Cells](#Cells)
* [Mapping](#Mapping)
* [Data Distribution](#Distribution)
* [Cluster](#Cluster)
* [Genes](#Genes)

### 18011

Estimated Number of Cells

### 1221

Median Gene per Cell

### 4756

Mean Reads per Cell

### 1763

Median UMI per Cell

#### Summary

Number of Input Reads: The number of total raw reads of the sample.

Reads With Valid Barcodes: The number and fraction of reads with valid barcodes.

Sequencing Saturation: Sequencing saturation refers to the percentage of reads originating from a duplicate UMI
(in other words, an mRNA molecule that was sequenced more than 1 time).

Q30 Bases in CB+UMI: The fraction of cell-barcode and UMI bases with Q-scores >= 30.

Q30 Bases in RNA read: The fraction of RNA read bases with Q-scores >= 30.

Mixture Ratio: The fraction of cell-associated barcodes estimated to be associated with more than one cell
(This value is calculated by the proportion of cell-barcode with less than 80% reads and larger than 20% reads mapped to two different genome,
so the value is only meaningful when mapping the reference genomes of two different species).

Sequencing Saturation Plot: This plot shows the Sequencing Saturation metric as a function of downsampled sequencing depth,
up to the observed sequencing depth. A high saturation indicates we are detecting the vast majority of mRNA molecules in the samples,
and thus don't need to sequence the libraries any deeper.

|  |  |
| --- | --- |
| Sample ID | PBMC\_20250321 |
| Reference |  |
| Kit | kitv1 |
| Pipeline version | Cell Sketcher v1.1 |
| Include introns | True |
| Number of Input Reads | 387482792 |
| Reads With Valid Barcodes | 342403776 (88.366%) |
| Sequencing Saturation | 0.547953 |
| Q30 Bases in CB+UMI | 0.948288 |
| Q30 Bases in RNA read | 0.920987 |
| Mixture Ratio | - |

#### Cells

Estimated Number of Cells: The number of barcodes evaluated as cells.

Confidently Mapped Reads in Cells: The number and fraction of reads with cell-associated barcodes and uniquely mapped to the transcriptome.

Mean Reads per Cell: The average number of mapped reads per cell detected.

Median Reads per Cell: The median number of unique mapped reads per cell detected.

Mean UMI per Cell: The average number of UMIs per cell detected.

Median UMI per Cell: The median number of UMIs per cell detected.

Mean Gene per Cell: The average number of Genes per cell detected.

Median Gene per Cell: The median number of Genes per cell detected.

Total Gene Detected: The number of genes with at least one UMI count in cell detected.

Barcode Rank Plot:  Total UMI count for each barcode plotted against its rank.
The distribution of total counts exhibits a sharp transition between barcodes with large and small total counts,
probably corresponding to cell-containing and empty droplets respectively.
The blue part of the curve indicates cell-associated barcodes, while the gray part may be the background-associated barcodes.

|  |  |
| --- | --- |
| Estimated Number of Cells | 18011 |
| Confidently Mapped Reads in Cells | 85663444 (0.3996) |
| Mean Reads per Cell | 4756 |
| Median Reads per Cell | 3959 |
| Mean UMI per Cell | 2122 |
| Median UMI per Cell | 1763 |
| Mean Gene per Cell | 1375 |
| Median Gene per Cell | 1221 |
| Total Gene Detected | 28487 |

#### Mapping

Reads Mapped to Genome: The number and fraction of reads mapped to the genome.

Reads Mapped Confidently to Genome: The number and fraction of reads uniquely mapped to the genome.

Reads Mapped Confidently to Transcriptome: The number and fraction of reads uniquely mapped to the transcriptome. The read must be consistent with annotated splice junctions.

Reads Mapped Confidently to Exonic Regions: The number and fraction of reads that uniquely mapped to an exonic region of the genome.

Reads Mapped Confidently to Intronic Regions: The number and fraction of reads that uniquely mapped to an intronic region of the genome.

Reads Mapped Confidently to Intergenic Regions: The number and fraction of reads that uniquely mapped to the intergenic region of the genome.

Reads Mapped Antisense to Gene: The number and fraction of reads uniquely mapped to the antisense strand of gene.

|  |  |
| --- | --- |
| Reads Mapped to Genome | 335690482 (86.634%) |
| Reads Mapped Confidently to Genome | 272759464 (70.393%) |
| Reads Mapped Confidently to Transcriptome | 243516546 (62.846%) |
| Reads Mapped Confidently to Exonic Regions | 136684801 (40.718%) |
| Reads Mapped Confidently to Intronic Regions | 179764005 (53.551%) |
| Reads Mapped Confidently to Intergenic Regions | 19241676 (5.732%) |
| Reads Mapped Antisense to Gene | 0 (0.0%) |

#### Data Distribution

Gene counts: The distribution of effective gene counts detected in each cell.

UMI counts: The distribution of UMI counts detected in each cell.

Mito percentage: The distribution of mitochondrial fraction detected in each cell.

#### UMAP Cluster

The display is limited to a random subset of cells.

left: This plot shows the UMI counts for each cell barcode.
Each dot associated with a cell barcode and is colored by the number of total UMI counts.
The coordinate axes represents the 2-dimensional embedding produced by the UMAP(Uniform Manifold Approximation and Projection)algorithm.

right: This plot shows the automated clustering result for each cell-barcode by UMAP algorithm.
Each dot associated with a cell barcode and is colored according to different cluster.

#### Top Differentially Expressed Genes

Table: The table shows the top 30 differentially expressed genes for each cluster.
Here a differential expression test was performed between each cluster and the rest of the sample for each feature.
The avg\_log2FC is an estimate of the log2 ratio of expression in a cluster to that in all other cells.
The p\_val is a measure of the statistical significance of the expression difference and the p\_val\_adj is adjusted p-value, based on bonferroni correction using all features in the dataset.

Bubbleplot: This figure shows the expression distribution of top5 gene in each cluster.
The color represents the average expression of gene in different clusters,
and the bubble size represents the proportion of gene expression in cells in different clusters.

![](data:image/png;base64...)